qp_klp/Assays.py

```diff
@@ -0,0 +1,511 @@
+
+                if 'target_gene' in metadata[list(metadata.keys())[0]]:
+                    tg = metadata[list(metadata.keys())[0]]['target_gene']
+                    for key in Amplicon.AMPLICON_SUB_TYPES:
+                        if key in tg:
+                            data['data_type'] = key
+
+                    if data['data_type'] is None:
+                        raise ValueError("data_type could not be "
+                                         "determined from target_gene "
+                                         "column")
+                else:
+                    raise ValueError("target_gene must be specified for "
+                                     "amplicon type")
+
+                reply = self.qclient.post('/qiita_db/prep_template/',
+                                          data=data)
+                prep_id = reply['prep']
+                results[study_id].append((prep_id, afact_name, is_repl))
+                self.run_prefixes[prep_id] = [metadata[sample]['run_prefix']
+                                              for sample in metadata]
+
+        self.touched_studies_prep_info = results
+        return results
+
+    def load_preps_into_qiita(self):
+        data = []
+        for project, _, qiita_id in self.special_map:
+            fastq_files = self._get_postqc_fastq_files(
+                self.pipeline.output_path, project)
+
+            for vals in self.touched_studies_prep_info[qiita_id]:
+                prep_id, artifact_name, is_repl = vals
+                if is_repl:
+                    # for Amplicon runs, each prep needs a copy of the
+                    # entire set of fastq files, because demuxing samples
+                    # happens downstream. If we don't make copies of the
+                    # files, Qiita will move the files when loading the
+                    # first prep and they won't be available for the
+                    # second prep and after.
+                    # Note that this will leave the original files present
+                    # in the working directory after processing instead of
+                    # being moved.
+                    working_set = self._copy_files(fastq_files)
+                else:
+                    working_set = fastq_files
+
+                data.append(self._load_prep_into_qiita(
+                    self.qclient, prep_id, artifact_name, qiita_id, project,
+                    working_set, ARTIFACT_TYPE_AMPLICON))
+
+        df = pd.DataFrame(data)
+        opath = join(self.pipeline.output_path, 'touched_studies.html')
+        with open(opath, 'w') as f:
+            f.write(df.to_html(border=2, index=False, justify="left",
+                               render_links=True, escape=False))
+
+        return df
+
+
+class MetaOmic(Assay):
+    """
+    MetaOmic() is a base class for Metagenomic() and Metatranscriptomic(),
+    which are currently identical in functionality.
+    """
+    assay_type = ASSAY_NAME_NONE
```

Comment:
This is a duplicate as Assay should already have this and might lead to confusions, no?

Reply:
Good catch, thank you!